src/components/Album/Album.js

```diff
@@ -1,39 +1,76 @@
-import React, { useState, useEffect, useContext } from "react"
-import { useParams } from "react-router-dom"
-import './Album.css'
-import Form from '../Form/Form'
-import { getAlbumsByMasterId } from "../../api"
-import UserContext from "../UserContext/UserContext"
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import { getAlbumDetails } from '../../api';
+import Form from '../Form/Form';
+import './Album.css'; 
 
 function Album() {
-  const {setCurrentUser} = useContext(UserContext)
-  const [album, setAlbum] = useState(null)
-  const [isLoading, setIsLoading] = useState(true)
-  const [modal, setModal] = useState(false)
-  
-  const {id} = useParams()
-  
-  const showModal = () => {
-    modal ? setModal(false) : setModal(true)
-  }
+  const { id } = useParams();
+  const [albumDetails, setAlbumDetails] = useState(null);
+  const [isLoading, setLoading] = useState(true);
+  const [modal, setModal] = useState(false);
 
   useEffect(() => {
-    getAlbumsByMasterId(id)
-      .then(res => {
-        setAlbum(res)
-        setIsLoading(false)
+    setLoading(true);
+    getAlbumDetails(id)
+      .then(data => {
+        setAlbumDetails(data);
+        setLoading(false);
       })
-  }, [])
-
-  
-  if(!isLoading) {
-    return (
-      <>
-       <p>{id} hello</p>
-       {!modal && <button onClick={showModal}>Add a journal entry</button>}
-       {modal && <Form {...album} setCurrentUser={setCurrentUser} showModal={showModal}/>}
-      </>
-    )
-  } 
+      .catch(error => {
+        console.error('Error fetching album details:', error);
+        setLoading(false);
+      });
+  }, [id]);
+
+  const handleAddToJournal = () => {
+    console.log('Album added to Journal!');
+  };
+
+  const handleAddToCollections = () => {
+    console.log('Album added to Collections!');
+  };
+
+  const showModal = () => {
+    setModal(prevModal => !prevModal);
+  };
+
+  if (isLoading) {
+    return <div>Loading...</div>;
+  }
+
+  if (!albumDetails) {
+    return <div>Album not found.</div>;
+  }
+
+  const { title, artist, releaseDate, genre, styles, tracklist, coverImg } = albumDetails;
+  return (
+    <div>
+      
+      <img src={coverImg} alt={`Cover art for ${title}`} />
+      <h2>{title}</h2>
+      <p>Artist: {artist}</p>
+      <p>Release Date: {releaseDate}</p>
+      <p>Genre: {genre}</p>
+      <p>Styles: {styles.join(', ')}</p>
+      {tracklist && tracklist.length > 0 && (
+        <>
+          <h3>Tracklist:</h3>
+          <ul>
```

Comment:
could we make this an ordered list?

Reply:
I think this should be the position from the API. It gives the side and the position in one. Then remove the styling to get rid of the buller

Reply:
Maybe also add the duration right after the song title?

Reply:
<h3>Tracklist:</h3>
    <ol> 
      {tracklist.map((track, index) => (
        <li key={index}>{track}</li>
      ))}
    </ol>
  </>
)}

changed to ordered list

Reply:
Changed <ul> to <ol>. !

Reply:
i may need to add a function to fetch additional information from the API that includes duration and position in the album. Was looking at endpoints trying to access these properties.